fms/models/llama.py

```diff
@@ -583,16 +633,40 @@ def _hf_sd_to_fms_sd(hf_sd: Mapping) -> Mapping:
         (r"post_attention_layernorm", "ff_ln"),
     ]
     new_sd = {}
-
-    trans_required_pattern = re.compile("layers.[0-9]+.attn.(query|key).(weight|bias)")
-    for name, param in hf_sd.items():
+    for name, param in input_sd.items():
         new_name = name
         for pattern, repl in replacements:
             new_name = re.sub(pattern, repl, new_name)
         new_sd[new_name] = param
+    return new_sd
+
+
+serialization.register_adapter_step("llama", "hf_to_fms_names", _hf_to_fms_names)
 
+
+def _hf_to_fms_rope(
+    input_sd: Mapping, extra_kwargs: Optional[Mapping] = None
+) -> Mapping:
+    new_sd = {}
+
+    if extra_kwargs and "model_config" in extra_kwargs:
```

Comment:
will there ever be a case where `model_config` is not passed into this adapter when architecture is llama?

Reply:
not on our scripts, but we expose these APIs publicly, so it's better to be careful about it